src/app/scripts/cac/control/cac-control-directions.js

```diff
@@ -178,35 +149,30 @@ CAC.Control.Directions = (function (_, $, Control, Geocoder, Routing, Typeahead,
         }
 
         if (mode.indexOf('BICYCLE') > -1) {
-            // TODO: re-enable bike optimization options when control re-implemented
-            //var bikeTriangleOpt = $('option:selected', options.selectors.bikeTriangleDiv);
-            //var bikeTriangle = bikeTriangleOpt.val();
-
-            //$.extend(otpOptions, {optimize: 'TRIANGLE'},
-            //         modeOptionsControl.options.bikeTriangle[bikeTriangle]);
-            //UserPreferences.setPreference('bikeTriangle', bikeTriangle);
+            // set bike trip optimization option
+            var bikeTriangle = UserPreferences.getPreference('bikeTriangle');
 
-            // allow longer bike riding when using public transit
-            $.extend(otpOptions, { maxWalkDistance: MAXWALK_BIKE * METERS_PER_MILE });
-        } else {
-            var maxWalk = $('input', options.selectors.maxWalkDiv).val();
-            if (maxWalk) {
-                UserPreferences.setPreference('maxWalk', maxWalk);
-                $.extend(otpOptions, { maxWalkDistance: maxWalk * METERS_PER_MILE });
+            if (_.has(modeOptionsControl.options.bikeTriangle, bikeTriangle)) {
+                $.extend(otpOptions, {optimize: 'TRIANGLE'},
+                     modeOptionsControl.options.bikeTriangle[bikeTriangle]);
             } else {
-                UserPreferences.setPreference('maxWalk', undefined);
+                console.error('unrecognized bike triangle option ' + bikeTriangle);
             }
 
-            // true if box checked
-            var wheelchair = $('input', options.selectors.wheelchairDiv).prop('checked');
-            UserPreferences.setPreference('wheelchair', wheelchair);
-            $.extend(otpOptions, { wheelchair: wheelchair });
+            // check user preference for bike share here, and update query mode if so
+            if (UserPreferences.getPreference('bikeShare')) {
+                mode = mode.replace('BICYCLE', 'BICYCLE_RENT');
```

Comment:
Once we get these preferences loading from the URL, is there a possibility that the mode will already include `BICYCLE_RENT` at this point?  If so this would turn it into `BICYCLE_RENT_RENT`.
```
mode = mode.split(',').map(function (m) {
    return m === 'BICYCLE' ? 'BICYCLE_RENT' : m;
}).join(',');
```
is a little wordier, but it's idempotent.